### PR TITLE
unleash-client-cpp: Add initial version 1.1.0

### PR DESCRIPTION
**Pull request by aruizs**
- Library to use the Unleash Feature Toggle service.

Specify library name and version:  **unleash-client-cpp/1.1.0**

I am the author of the unleash-client-cpp library.
External dependencies: cpr and nlohmann json libraries.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/unleash-client-cpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (uilianries) [inline recipes/unleash-client-cpp/all/conanfile.py:79]:
```suggestion
```

Not needed. CMake helper already solves it. Same for INDENPENDENT_POSITION_CODE

File: recipes/unleash-client-cpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["unleash"]
+        self.cpp_info.names["cmake_find_package"] = "unleash"
```
Comment (uilianries) [inline recipes/unleash-client-cpp/all/conanfile.py:98]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "unleash"
        self.cpp_info.names["cmake_find_package_multi"] = "unleash"
```

File: recipes/unleash-client-cpp/all/patches/cmake-install.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
+diff --git a/packaging/CMakeLists.txt b/packaging/CMakeLists.txt
```
Comment (uilianries) [inline recipes/unleash-client-cpp/all/patches/cmake-install.patch:1]: Don't need to patch it, avoid patching if you can. It will be better removing the cmake folder after install step, instead creating a patch.

File: recipes/unleash-client-cpp/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/unleash-client-cpp/all/test_package/CMakeLists.txt:8]:
```suggestion
conan_basic_setup(TARGETS)

find_package(unleash CONFIG REQUIRED)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} unleash::unleash)
```
To validate the cmake target generated

File: recipes/unleash-client-cpp/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/unleash-client-cpp/all/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

To generate the cmake config file.

File: recipes/unleash-client-cpp/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (uilianries) [inline recipes/unleash-client-cpp/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```
No longer needed

File: recipes/unleash-client-cpp/all/conanfile.py (hunk 1 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/conanfile.py]: Done.

File: recipes/unleash-client-cpp/all/conanfile.py (hunk 2 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/conanfile.py:98]: Done.

Comment (aruizs) [reply recipes/unleash-client-cpp/all/conanfile.py:98]: Done.

File: recipes/unleash-client-cpp/all/patches/cmake-install.patch (hunk 3 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/patches/cmake-install.patch:1]: Done.

File: recipes/unleash-client-cpp/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/test_package/CMakeLists.txt:8]: Done.

File: recipes/unleash-client-cpp/all/test_package/conanfile.py (hunk 5 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/test_package/conanfile.py]: Done.

File: recipes/unleash-client-cpp/all/test_package/conanfile.py (hunk 6 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/test_package/conanfile.py]: Done.

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Comment by aruizs** [comment]
> Please do not force push pray GitHub forces us to restart the review which is not fun!

Ok, sorry. I just wanted to have a clean commit history.

**Comment by prince-chrismc** [comment]
The bot will squash before merging 👌

File: recipes/unleash-client-cpp/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    requires = (
+        "cpr/1.7.2",
+        "nlohmann_json/3.10.5",
+    )
+
+    build_requires = "cmake/3.22.0"
```
Comment (prince-chrismc) [inline recipes/unleash-client-cpp/all/conanfile.py:33]: Why is this required 🤔?

File: recipes/unleash-client-cpp/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (prince-chrismc) [reply recipes/unleash-client-cpp/all/test_package/CMakeLists.txt:8]: This comment seems to be missing

File: recipes/unleash-client-cpp/all/conanfile.py (hunk 7 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/conanfile.py:33]: I chose that cmake version to match the opensll version in the rest of dependencies. Another option would be to add the openssl explicitily in the recipe with an specific version.

File: recipes/unleash-client-cpp/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/test_package/CMakeLists.txt]: You're right, thanks. Now it is applied.

File: recipes/unleash-client-cpp/all/conanfile.py (hunk 7 above)
Comment (aruizs) [reply recipes/unleash-client-cpp/all/conanfile.py]: Anyway, I've updated the unleash-client-cpp library to reduce the minimum cmake required version.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jgsogo — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
This library downloads and call cmake_conan at build time:
https://github.com/aruizs/unleash-client-cpp/blob/v1.1.1/CMakeLists.txt#L28-L29
https://github.com/aruizs/unleash-client-cpp/blob/v1.1.1/cmake/Conan.cmake

It must be removed.
